### ISSUE TITLE
Assembly Patching

### ISSUE THREAD

**Issue by Knaackee** [StevenLiekens/dotnet-core-appveyor#5]
Do someone know if Assembly Patching does work? Cant get it work.

Thanks in advance!

**Comment by sliekens** [comment]
No, all assembly info is generated based on what it says in the project.json files. The assembly patching in AppVeyor runs before any assembly info code generation takes place.

**Comment by Knaackee** [comment]
Hmm, ok. So does that mean you cant have an auto increment of the verions number besides the version prefix ? 

Thanks for your help

**Comment by sliekens** [comment]
I'm not sure but I think you can control the version info using environment variables
- `DOTNET_BUILD_VERSION`: can be used instead of setting --version-suffix on the cmdline
- `DOTNET_ASSEMBLY_FILE_VERSION`: has no cmdline equivalent?

It's not clear if `DOTNET_BUILD_VERSION` should contain only the suffix or the entire version string.

You might try something like this:

``` powershell
ps: $Env:DOTNET_BUILD_VERSION = "1.0.$Env:APPVEYOR_BUILD_NUMBER-CI"
ps: $Env:DOTNET_ASSEMBLY_FILE_VERSION = "1.0.$Env:APPVEYOR_BUILD_NUMBER"
```

**Comment by Knaackee** [comment]
I test that out and report back soon.

**Comment by Knaackee** [comment]
I am getting 

The assembly file version is invalid: 1.0..51

**Comment by sliekens** [comment]
I forgot that assembly versions have 4 components instead of 3. It's interesting that dotnet parses it as `1.0.<empty>.51`.

Try setting all 4 components: `1.0.0.$Env:APPVEYOR_BUILD_NUMBER`

**Comment by Knaackee** [comment]
already tried it with no luck :) 

I found where the exception happend

https://github.com/borgdylan/dotnet-cli/blob/master/src/Microsoft.DotNet.ProjectModel/ProjectReader.cs#L139

**Comment by sliekens** [comment]
@dknaack did you figure out a solution?
